src/main.zig

```diff
@@ -34,8 +34,13 @@ pub const std_options = struct {
         const level_txt = comptime level.asText();
         const scope_txt = comptime @tagName(scope);
 
-        std.debug.print("{s:<5}: ({s:^6}): ", .{ level_txt, if (comptime std.mem.startsWith(u8, scope_txt, "zls_")) scope_txt[4..] else scope_txt });
-        std.debug.print(format ++ "\n", args);
```

Comment:
```
std.debug.print("{s:<5}: ({s:^6}): " ++ fmt ++ "\n", .{ level_txt, if (comptime std.mem.startsWith(u8, scope_txt, "zls_")) scope_txt[4..] else scope_txt } ++ args);
```

Reply:
The reason why I wrote it this way is because it creates less and simpler function instantiations of `std.fmt.format`.

```zig
if (comptime build_options.opt) {
    const stderr = std.io.getStdErr().writer();
    std.debug.getStderrMutex().lock();
    defer std.debug.getStderrMutex().unlock();

    stderr.print("{s:<5}: ({s:^6}): ", .{ level_txt, if (comptime std.mem.startsWith(u8, scope_txt, "zls_")) scope_txt[4..] else scope_txt }) catch return;
    stderr.print(format, args) catch return;
    stderr.writeByte('\n') catch return;
} else {
    std.debug.print("{s:<5}: ({s:^6}): " ++ format ++ "\n", .{ level_txt, if (comptime std.mem.startsWith(u8, scope_txt, "zls_")) scope_txt[4..] else scope_txt } ++ args);
}
```

```
Benchmark 1 (9 runs): zig build-exe src/main.zig # where build_options.opt is false
  measurement          mean ± σ            min … max           outliers         delta
  wall_time          14.7s  ± 89.0ms    14.7s  … 14.9s           0 ( 0%)        0%
  peak_rss           1.06GB ±  175KB    1.06GB … 1.06GB          0 ( 0%)        0%
  cpu_cycles         50.0G  ±  102M     49.8G  … 50.1G           0 ( 0%)        0%
  instructions       63.1G  ± 12.7M     63.1G  … 63.1G           0 ( 0%)        0%
  cache_references    888M  ± 1.37M      887M  …  890M           0 ( 0%)        0%
  cache_misses       43.4M  ±  364K     42.9M  … 44.0M           0 ( 0%)        0%
  branch_misses       346M  ±  481K      345M  …  347M           0 ( 0%)        0%
Benchmark 2 (9 runs): zig build-exe src/main.zig ... # where build_options.opt is true
  measurement          mean ± σ            min … max           outliers         delta
  wall_time          14.5s  ± 21.2ms    14.5s  … 14.6s           0 ( 0%)          -  1.4% ±  0.4%
  peak_rss           1.05GB ±  191KB    1.05GB … 1.05GB          1 (11%)        ⚡-  1.1% ±  0.0%
  cpu_cycles         49.4G  ± 86.3M     49.3G  … 49.5G           0 ( 0%)          -  1.2% ±  0.2%
  instructions       62.4G  ± 15.6M     62.4G  … 62.4G           0 ( 0%)        ⚡-  1.1% ±  0.0%
  cache_references    875M  ± 2.74M      871M  …  880M           0 ( 0%)        ⚡-  1.5% ±  0.2%
  cache_misses       42.7M  ±  307K     42.4M  … 43.2M           0 ( 0%)          -  1.5% ±  0.8%
  branch_misses       343M  ±  363K      343M  …  344M           0 ( 0%)          -  0.8% ±  0.1%
```

The zls binary also appeared to be 14kb smaller when compiling with ReleaseSmall.